OpenRA.Mods.Common/Traits/Upgrades/UpgradeManager.cs

```diff
@@ -51,7 +51,7 @@ class UpgradeState
 
 		readonly List<TimedUpgrade> timedUpgrades = new List<TimedUpgrade>();
 		readonly Lazy<Dictionary<string, UpgradeState>> upgrades;
-		readonly Dictionary<IUpgradable, int> levels = new Dictionary<IUpgradable, int>();
+		readonly Dictionary<IUpgradable, Dictionary<string, int>> levels = new Dictionary<IUpgradable,Dictionary<string,int>>();
```

Comment:
OpenRA.Mods.Common/Traits/Upgrades/UpgradeManager.cs:L54: [SA1001] Invalid spacing around the comma.